activity-module

brandonschabell opened the pull request:
Implemented route tracking, pacing, distance tracking, and calorie counting.
Also contains multiDex support in the build.gradle file

Commit:
Created Activity Module!

Yay!

Commit:
Merge branch 'master' into activity-module

Commit:
Implemented Google Map API and fixed build.gradle issues. Using "release" build

Commit:
Implemented Google Map API and fixed build.gradle issues. Using "release" build

Commit:
Implemented Google Map API and fixed build.gradle issues. Using "release" build

Commit:
Added *some* of the location functionality. Has not been tested to see if it runs.

Commit:
Made things prettier. Still lacking "current location" functionality

Commit:
Location is working correctly. Map is able to get my exact location and request permissions if necessary.

Commit:
Merge remote-tracking branch 'origin/activity-module' into activity-module

# Conflicts:
#	Health/src/main/AndroidManifest.xml
#	Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java

Commit:
Location is working correctly. Map is able to get my exact location and request permissions if necessary.

Commit:
Polyline added, however live tracking is not enabled yet.

Commit:
Live tracking is enabled, a polyline is drawn, and current location is shown and constantly updated.

Commit:
Distance and Timestamps are being calculated (accuracy has not been tested)

Commit:
Sprint 1 complete! Calorie counting works along with distance, pacing, and route tracking

avaidyam commented on Health/src/main/java/com/sciencesquad/health/activity/ActivityModule.java:
```diff
@@ -0,0 +1,5 @@
+package com.sciencesquad.health.activity;
+
+public class ActivityModule {
```
lol well ok, fuck me then

avaidyam commented on Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:
```diff
@@ -0,0 +1,326 @@
+    public double calorieBurn(double speed, double timeDiff, double weightKG) {
+        // METS List (Metric for the exertion for each exercise)
+        double METS = 0;
+        double speedMPH = speed*2.23694; //converting speed from m/s to MPH
```
There _must_ be a better way to do this than a bunch of if statements...

avaidyam commented on Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:
```diff
@@ -0,0 +1,326 @@
+        switch (requestCode) {
+            case REQUEST_LOCATION_PERMISSION: {
+                if ((grantResults.length > 0) && (grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+                    // Do what you wanted to do with the permissions
```
So.... why is this here?

avaidyam commented on Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:
```diff
@@ -0,0 +1,326 @@
+    public void onConnected(Bundle bundle) {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION}, REQUEST_LOCATION_PERMISSION);
+            // here to request the missing permissions, and then overriding
```
Why not do what the comment says?

brandonschabell replied:
that's what the thing above is doing

avaidyam replied:
so very lost, but we'll go over it next time, no worries. :crown:

Commit:
Sprint 1 complete! Calorie counting works along with distance, pacing, and route tracking

Commit:
Merge branch 'master' into activity-module

# Conflicts:
#	Health/src/main/java/com/sciencesquad/health/MainActivity.java
#	Health/src/main/res/menu/activity_health_main_drawer.xml
#	Health/src/main/res/values/strings.xml

Commit:
Started to implement a check for "significant distance" traveled.

Commit:
Merge remote-tracking branch 'origin/activity-module' into activity-module

Commit:
Successful Builds

Resolved some silly issues, everything looks good from this branch (as
far as building gradle & project)!!!!

Commit:
Tweaked the check for "significant distance" traveled.

Commit:
Merge remote-tracking branch 'origin/activity-module' into activity-module

# Conflicts:
#	Health/src/main/AndroidManifest.xml

colinrford commented:
Dear anyone looking, 
@brandonschabell said that there's a local file that doesn't sync with git. Due to this, Travis CI will fail. However, this branch works just fine locally for us (so it seems?????????)

Brahmancer commented:
It looks like his branch is a few commits behind master. Maybe that's the reason?

colinrford commented:
@Brahmancer we just merged master and resolved conflicts, I'm 99.9999% sure it's not behind master

Commit:
Tweaked the check for "significant distance" traveled.

Brahmancer commented:
![screen shot 2016-03-04 at 12 13 13 am](https://cloud.githubusercontent.com/assets/5718941/13518625/12a97d64-e19e-11e5-94e4-bd0fc6a1f016.png)

That's what I'm seeing.

colinrford commented:
Github is pissing me off because we literally resolved all issues

Commit:
Merge branch 'master' into activity-module

hopefully this works this time

colinrford commented:
@Brahmancer you were right, for some reason it didn't show up for me? I assume I was doing something wrong then. This branch should be caught up now commit-wise.

colinrford commented:
@avaidyam I'm going to leave this branch up to you to merge unless there's something I can do. Brandon said it wasn't going to pass Travis because of the Health365.jks file

Brahmancer commented:
@colinrford were you able to build it on your machine?

colinrford commented:
@Brahmancer I was able to build this on my computer, however, I did not test it on my phone. Brandon had it working, but perhaps we can double check with him before we proceed with merging

Brahmancer commented:
I tried running this branch on my end and it gives me the thing below. As unfortunate/insecure as it is, we need that keystore in our repo. 
Error:Execution failed for task ':Health:validateConfigSigning'.

> Keystore file /Users/brandonschabell/keystores/Health365.jks not found for signing config 'config'.

colinrford commented:
@brandonschabell suggested we each keep it locally and adjust the path accordingly

Brahmancer commented on Health/build.gradle:
```diff
@@ -21,9 +21,16 @@ repositories {
+            storeFile file('/Users/brandonschabell/keystores/Health365.jks')
+            storePassword 'd3012356101530'
+        }
+    }
```
Trying removing this Part 1

Brahmancer commented:
What that is implying is that each of us set up a gradle.properties in the ~/.gradle directory and create environment variables. Here's a link on how to set that up. 
http://stackoverflow.com/questions/18328730/how-to-create-a-release-signed-apk-file-using-gradle/21020469#21020469

However, if we do it that way, we need to set up a gradle.properties for Travis which I'm not sure how to do.

colinrford commented:
That's something we should probably do, although I don't know if it should happen today. That's a good find though!

Brahmancer commented:
For real though, with time constraints, not sure how to approach this situation without being insecure.

colinrford commented:
We could leave this PR for when we're all around, or for Aditya to override. Then whoever demos the app or this aspect of the app will make sure they have the .jpk file n all that

Brahmancer commented:
We'll talk about it more when we get together.

Commit:
Fixed some bugs relating to the path drawing. Also made minimum distance resolution dependent on GPS accuracy. Uses 1-sigma radius as a minimum distance to trigger a change in position.

Commit:
Merge remote-tracking branch 'origin/activity-module' into activity-module

avaidyam commented:
Override it is.

Commit:
Merge branch 'master' into activity-module

# Conflicts:
#	Health/build.gradle
